x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+      );
+      const isEnabled = await actionMenuViewInLensButton.isEnabled();
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Delete index pattern field" action menu button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
+    }
+    public async clickActionMenuDeleteIndexPatternFieldButton(fieldName: string) {
+      const testSubj = 'dataVisualizerActionDeleteIndexPatternFieldButton';
+      await retry.tryForTime(5000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+
+        const button = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+        );
+        await button.click();
+        await this.ensureActionsMenuClosed(fieldName, testSubj);
+      });
+    }
+
+    public async assertActionMenuExists(fieldName: string) {
+      const actionButton = this.rowSelector(fieldName, 'euiCollapsedItemActionsButton');
+      await testSubjects.existOrFail(actionButton);
+    }
+
+    public async assertViewInLensActionEnabled(fieldName: string, expectedValue: boolean) {
       const actionButton = this.rowSelector(fieldName, 'dataVisualizerActionViewInLensButton');
       await testSubjects.existOrFail(actionButton);
-      await testSubjects.isEnabled(actionButton);
+      const isEnabled = await testSubjects.isEnabled(actionButton);
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Edit index pattern" button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
     }
 
     public async assertViewInLensActionNotExists(fieldName: string) {
       const actionButton = this.rowSelector(fieldName, 'dataVisualizerActionViewInLensButton');
       await testSubjects.missingOrFail(actionButton);
     }
 
+    public async assertEditIndexPatternFieldButtonEnabled(
+      fieldName: string,
+      expectedValue: boolean
+    ) {
+      const selector = this.rowSelector(
+        fieldName,
+        'dataVisualizerActionEditIndexPatternFieldButton'
+      );
+      await testSubjects.existOrFail(selector);
+      const isEnabled = await testSubjects.isEnabled(selector);
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Edit index pattern" button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
+    }
+
+    public async clickEditIndexPatternFieldButton(fieldName: string) {
+      await retry.tryForTime(5000, async () => {
+        await this.assertEditIndexPatternFieldButtonEnabled(fieldName, true);
+        await testSubjects.click(
+          this.rowSelector(fieldName, 'dataVisualizerActionEditIndexPatternFieldButton')
+        );
```

Comment:
We should assert that the click actually did something, e.g. check that the edit field flyout is open.

Reply:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e